src/query/useSimulateTx.ts

```diff
@@ -0,0 +1,62 @@
+import { useMutation } from "@tanstack/react-query";
+import { AnyObject } from "@/types/types";
+
+type SimulateTxProps = {
+  rpcUrl: string;
+  transactionXdr: string;
+  instructionLeeway?: string;
+};
+
+export const useSimulateTx = () => {
+  const mutation = useMutation({
+    mutationFn: async ({
+      rpcUrl,
+      transactionXdr,
+      instructionLeeway,
+    }: SimulateTxProps) => {
+      const res = await fetch(rpcUrl, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          jsonrpc: "2.0",
+          id: 1,
+          method: "simulateTransaction",
+          params: {
+            transaction: transactionXdr,
+            ...(instructionLeeway
```

Comment:
oh I might do it this way for rpc endpoints instead of my current approach to `instructionLeeway: Number(params.resourceConfig) || undefined`. TY!

Reply:
:+1: This only works if the server is not expecting that param when optional. Sometimes we still need to pass the param without value. Just something to keep in mind.